router/batchrouter/asyncdestinationmanager/sftp/utils.go

```diff
@@ -0,0 +1,323 @@
+			row = append(row, fmt.Sprintf("%v", fields[key]))
+		}
+		if err := writer.Write(row); err != nil {
+			return "", err
+		}
+	}
+
+	// Flush any buffered data to the underlying writer
+	writer.Flush()
+	if err := writer.Error(); err != nil {
+		return "", err
+	}
+
+	return tmpFilePath, nil
+}
+
+func getTempFilePath() (string, error) {
+	localTmpDirName := fmt.Sprintf(`/%s/`, misc.RudderAsyncDestinationLogs)
+	tmpDirPath, err := misc.CreateTMPDIR()
+	if err != nil {
+		return "", err
+	}
+	folderPath := path.Join(tmpDirPath, localTmpDirName)
+	_, e := os.Stat(folderPath)
+	if os.IsNotExist(e) {
+		folderPath, _ = os.MkdirTemp(folderPath, "")
+	}
+	tmpFilePath := filepath.Join(folderPath, uuid.NewString())
+	return tmpFilePath, nil
+}
+
+func getUploadFilePath(path string) string {
+	// Define a regular expression to match dynamic variables
+	re := regexp.MustCompile(`{([^}]+)}`)
+
+	// Get the current date and time
+	now := time.Now()
+	// Replace dynamic variables with their actual values
+	result := re.ReplaceAllStringFunc(path, func(match string) string {
+		switch match {
+		case "{YYYY}":
+			return strconv.Itoa(now.Year())
+		case "{MM}":
+			return fmt.Sprintf("%02d", now.Month())
+		case "{DD}":
+			return fmt.Sprintf("%02d", now.Day())
+		case "{hh}":
+			return fmt.Sprintf("%02d", now.Hour())
+		case "{mm}":
+			return fmt.Sprintf("%02d", now.Minute())
+		case "{ss}":
+			return fmt.Sprintf("%02d", now.Second())
+		case "{ms}":
+			return fmt.Sprintf("%03d", now.Nanosecond()/1e6)
+		case "{timestampInSec}":
+			return strconv.FormatInt(now.Unix(), 10)
+		case "{timestampInMS}":
+			return strconv.FormatInt(now.UnixNano()/1e6, 10)
+		default:
+			// If the dynamic variable is not recognized, keep it unchanged
+			return match
+		}
+	})
+
+	return result
+}
+
+func generateErrorOutput(err string, importingJobIds []int64, destinationID string) common.AsyncUploadOutput {
+	return common.AsyncUploadOutput{
+		DestinationID: destinationID,
+		AbortCount:    len(importingJobIds),
+		AbortJobIDs:   importingJobIds,
+		AbortReason:   err,
+	}
+}
+
+func validate(d destConfig) error {
+	if d.AuthMethod == "" {
+		return errors.New("authMethod cannot be empty")
+	}
+
+	if d.Username == "" {
+		return errors.New("username cannot be empty")
+	}
+
+	if d.Host == "" {
+		return errors.New("host cannot be empty")
+	}
+
+	if err := isValidPort(d.Port); err != nil {
+		return err
+	}
+
+	if d.AuthMethod == "passwordAuth" && d.Password == "" {
+		return errors.New("password is required for password authentication")
+	}
+
+	if d.AuthMethod == "keyAuth" && d.PrivateKey == "" {
+		return errors.New("private key is required for key authentication")
+	}
+
+	if err := isValidFileFormat(d.FileFormat); err != nil {
+		return err
+	}
+
+	if d.FilePath == "" {
+		return errors.New("filePath cannot be empty")
+	}
+
+	return nil
+}
+
+func isValidPort(p string) error {
+	if p == "" {
+		return errors.New("port cannot be empty")
+	}
```

Comment:
We don't need this. `strconv.Atoi` already takes care of it.

Reply:
addressed